src/Parsers/ParserModifyEngineQuery.cpp

```diff
@@ -0,0 +1,67 @@
+#include <Common/typeid_cast.h>
+#include <Parsers/ParserModifyEngineQuery.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ExpressionListParsers.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/ParserPartition.h>
+#include <Parsers/ParserSelectWithUnionQuery.h>
+#include <Parsers/ParserSetQuery.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTIndexDeclaration.h>
+#include <Parsers/ASTModifyEngineQuery.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/parseDatabaseAndTableName.h>
+
+
+namespace DB
+{
+
+bool ParserModifyEngineQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    auto query = std::make_shared<ASTModifyEngineQuery>();
+    node = query;
+
+    ParserKeyword s_alter_table("ALTER TABLE");
```

Comment:
Should it be a variant of ALTER query then? See `ParserAlterQuery`, `ASTAlterQuery`, `InterpreterAlterQuery`

Reply:
I abandoned this idea when I realized that the conversion implementation does not fit well with the alter query interpreter, we can't just add a new alter command and execute it along with the rest, because in the process the table will be detached and even its class will change to StorageReplicatedMergeTree. 
But I think we can still use the parser and ast.
There is also an idea to redo this into a System query due to it's differences with other alter queries.